### PR TITLE
remove nose dependancy

### PR DESCRIPTION
**Pull request by Adersh97**

### CODE REVIEW

File: tests/test_vsa.py [hunk 1]
```diff
@@ -392,110 +382,108 @@ def is_equal(ast_0, ast_1):
...
+    assert is_equal(si_concat, si)
 
     # Concatenating two SIs that are of different sizes
     si_1 = SI(bits=64, stride=1, lower_bound=0, upper_bound=0xffffffffffffffff)
     si_2 = SI(bits=32, stride=1, lower_bound=0, upper_bound=0xffffffff)
     si_concat = si_1.concat(si_2)
-    nose.tools.assert_true(is_equal(si_concat, SI(bits=96, stride=1,
-                                                  lower_bound=0,
-                                                  upper_bound=0xffffffffffffffffffffffff)))
+    assert is_equal(si_concat, SI(bits=96, stride=1,                                                   lower_bound=0,                                                   upper_bound=0xffffffffffffffffffffffff))
```
Comment (twizmwazin) [inline tests/test_vsa.py:418]: Formatting

File: tests/test_vsa.py [hunk 2]
```diff
@@ -541,512 +529,79 @@ def VS(name=None, bits=None, region=None, val=None):
...
-    nose.tools.assert_true(claripy.backends.vsa.is_true(vsa_model(if_1.ite_excavated.args[1]) == vsa_model(VS(region='global', bits=32, val=0))))
-    nose.tools.assert_true(claripy.backends.vsa.is_true(vsa_model(if_1.ite_excavated.args[2]) == vsa_model(vs_2)))
+    assert claripy.backends.vsa.is_true(vsa_model(if_1.ite_excavated.args[1]) == vsa_model(VS(region='global', bits=32, val=0)))
+    assert claripy.backends.vsa.is_true(vsa_model(if_1.ite_excavated.args[2]) == vsa_model(vs_2))
 
     # if_2 = And(VS_3, IfProxy(si != 0, 0, 1)
     vs_3 = VS(region='global', bits=32, val=0xDEADCA7)
     si = claripy.SI(bits=32, stride=1, lower_bound=0, upper_bound=1)
     if_2 = (vs_3 & claripy.If(si != 0, claripy.SI(bits=32, stride=0, lower_bound=0, upper_bound=0), claripy.SI(bits=32, stride=0, lower_bound=0xffffffff, upper_bound=0xffffffff)))
-    nose.tools.assert_true(claripy.backends.vsa.is_true(vsa_model(if_2.ite_excavated.args[1]) == vsa_model(VS(region='global', bits=32, val=0))))
```
Comment (twizmwazin) [inline tests/test_vsa.py:620]: Looks like everything was deleted from this point down?

**Review by twizmwazin — CHANGES_REQUESTED** [review]
Please take a look at the inline comments and the CI failures. Also, for future PRs, you may want to choose different branch names. The CI will use branches from other repos if their names match, so I presume some of the failures in this CI run are actually issues in the angr PR.

File: tests/test_expression.py [hunk 3]
```diff
@@ -1,536 +1,519 @@
...
+        solver.add(a == -4)
+        assert list(solver.eval(a >> 1, 2)) == [2**32-2]
+
+    def test_bool_conversion(self):
+        a = claripy.BVV(42, 32)
         try:
-            b.convert(x+x+x+x)
-        except claripy.BackendError:
+            assert a == 42
+            assert False, "`assert ast` should raise an exception"
```
[the remark this answers is not in the document]
Comment (Adersh97) [reply tests/test_expression.py:501]: I don't know if the test case always raises claripy.BackendError.

[the remark this answers is not in the document]
Comment (rhelmot) [reply tests/test_expression.py:501]: They should always raise the error.

[the remark this answers is not in the document]
Comment (Adersh97) [reply tests/test_expression.py:501]: okay cool, I will use assertRaises then.

**Comment by twizmwazin** [comment]
Closing in favor of #261

**Pull request closed without merging**